javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/AbstractJavaParserContext.java

```diff
@@ -119,6 +119,7 @@ public final Optional<Context> getParent() {
         // is not the same as the current node.
         while (notMethodNode instanceof MethodCallExpr
                 || notMethodNode instanceof FieldAccessExpr
+                || (notMethodNode instanceof ObjectCreationExpr && notMethodNode.hasScope())
```

Comment:
I'm not sure that the expression new A().new B() can be solved correctly.

Reply:
Please add a comment explaining simply and clearly why this has been added

Reply:
I have added a comment, and a unit test for the case you mentioned.

Reply:
In fact, I added two unit tests :)